lib/elixir/lib/record.ex

```diff
@@ -255,7 +255,13 @@ defmodule Record do
   """
   defmacro defrecord(name, tag \\ nil, kv) do
     quote bind_quoted: [name: name, tag: tag, kv: kv] do
+      if Module.defines?(__MODULE__, {name, 0}) do
```

Comment:
this is somewhat pedantic, but defrecord defines 0, 1 and 2 arity defs so perhaps we should handle all of them? For example, if there's a 1-arity def already, we'll get a bad error too:

```
defmodule M do
  defmacro a(_) do
  end

  require Record
  Record.defrecord :a, [:a]
end
```

```
warning: defmacro a/1 has multiple clauses and also declares default values. In such cases, the default values should be defined in a header. Instead of:

    def foo(:first_clause, b \\ :default) do ... end
    def foo(:second_clause, b) do ... end

one should write:

    def foo(a, b \\ :default)
    def foo(:first_clause, b) do ... end
    def foo(:second_clause, b) do ... end

  a.ex:6

warning: this clause cannot match because a previous clause at line 2 always matches
  a.ex:6
```

Reply:
Right, this would leave out `name/1` and `/2`. I'll push a try at that.